### ISSUE TITLE
[Bug] ea27f9c9ed28 broke reference validation

### ISSUE THREAD

**Issue by vrothberg** [GoogleCloudPlatform/container-diff#183]
ea27f9c9ed28 broke the reference validation so that all prefixes (e.g., `daemon://`) will end up with:
```
ERRO[0000] invalid reference format
```

The problem seems to be that `ref, err := docker.ParseReference("//" + image)` is executed prior to removing the prefix. I wonder why the integration tests did not catch that regression.

**Comment by vrothberg** [comment]
It seems like there's another problem.  After applying the fixes from #184, there's another error when analyzing `daemon://` images:

```
ERRO[0001] Error processing image: Invalid status code returned when fetching blob 404 
```

**Comment by dlorenc** [comment]
Ah whoops, looks like we don't have integration test coverage for "daemon://", only the naked style which defaults to remote://. Thanks for the fix.

**Comment by dlorenc** [comment]
OK, I think I tracked down the test issue and the code issue. Sending a PR in a moment.

**Comment by vrothberg** [comment]
@dlorenc Thanks for looking into this. I'm still fighting with #184, I will ping you there for review once Travis is green. I was working on a PR for #175 when noticing the regression

**Comment by vrothberg** [comment]
Fixed by #187
